src/overlays/actors/ovl_Bg_Jya_Zurerukabe/z_bg_jya_zurerukabe.c

```diff
@@ -21,23 +36,161 @@ const ActorInit Bg_Jya_Zurerukabe_InitVars = {
     (ActorFunc)BgJyaZurerukabe_Update,
     (ActorFunc)BgJyaZurerukabe_Draw,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Jya_Zurerukabe/func_8089B440.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Jya_Zurerukabe/func_8089B4C8.s")
+static s16 D_8089B9F0[4] = { 943, 1043, 1243, 1343 };
+
+static s16 D_8089B9F8[4] = { -1, 1, -1, 1 };
+
+static s16 D_8089BA00[4] = { 48, 48, 36, 36 };
+
+static f32 D_8089BA08[4] = { 8.0f, 8.0f, 10.0f, 10.0f };
+
+static s16 D_8089BA18[6][2] = {
+    { 0x0388, 0x0395 }, { 0x03EA, 0x03FF }, { 0x0454, 0x0467 },
+    { 0x04B4, 0x04C1 }, { 0x0518, 0x0528 }, { 0x0581, 0x0590 },
+};
+
+static s16 D_8089BA30[6] = {
+    0, 0, 1, 2, 2, 3,
+};
+
+static InitChainEntry sInitChain[] = {
+    ICHAIN_VEC3F_DIV1000(scale, 100, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneForward, 1000, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneScale, 1200, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneDownward, 1000, ICHAIN_STOP),
+};
+
+extern Gfx D_06012340[];
+extern UNK_PTR D_06012508;
+
+void func_8089B440(BgJyaZurerukabe* this, GlobalContext* globalCtx, void* arg2, DynaPolyMoveFlag flags) {
```

Comment:
```suggestion
void func_8089B440(BgJyaZurerukabe* this, GlobalContext* globalCtx, void* arg2, s32 flags) {
```
[cf this](https://github.com/zeldaret/oot/pull/400#discussion_r491705572)